src/main/groovy/DemoQA/AlertsFramesAndWindowsPage.groovy

```diff
@@ -0,0 +1,40 @@
+package DemoQA
+
+import geb.Page
+import org.openqa.selenium.Alert
+import org.openqa.selenium.support.ui.ExpectedConditions
+
+class AlertsFramesAndWindowsPage extends Page {
+    static at = {
+        getDriver().getCurrentUrl() == url
+        mainHeader.isDisplayed()
+        mainHeader.text() == "Alerts"
+    }
+
+    static content = {
+        mainHeader          {$(".main-header")}
+        confirmButton       {$("#confirmButton")}
+    }
+
+    static url = "https://demoqa.com/alerts"
+
+    void clickConfirmButton() {
+        confirmButton.click()
+    }
+
+    void acceptPopUpAlert() {
```

Comment:
This can be split into two methods - one will check if the alert is present and can be used in test assertions the other will switch to alert and click the button. This way you don't have code repetition with the method below.